server/lib/routes/post-metrics.js

```diff
@@ -19,19 +20,28 @@ module.exports = function () {
     process: function (req, res) {
       // don't wait around to send a response.
       res.json({ success: true });
+
       process.nextTick(function () {
-        var metrics = req.body || {};
-        metrics.agent = req.get('user-agent');
+        try {
+          var metrics = req.body || {};
 
-        if (metrics.isSampledUser) {
-          metricsCollector.write(metrics);
-          // send the metrics body to the StatsD collector for processing
-          statsd.write(metrics);
-        }
+          var contentType = req.get('content-type') || '';
+          if (contentType.indexOf('text/plain') === 0) {
+            metrics = JSON.parse(metrics);
```

Comment:
the `metrics` inside of the `JSON.parse` should be `req.body` in case `req.body` is `''` and the request was sent with `Content-Type: text/plain`